pkg/translator/sumoicmarshaler/sumo_ic_marshaler.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumoicmarshaler
+
+import (
+	"encoding/json"
+
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const (
+	SourceCategoryKey = "_sourceCategory"
+	SourceHostKey     = "_sourceHost"
+	SourceNameKey     = "log.file.path_resolved"
+)
+
+type SumoICMarshaler struct {
+	format string
+}
+
+type fieldSet struct {
+	attributes map[string]interface{}
+}
+
+type icRecord struct {
+	Data           string   `json:"data"`
+	SourceName     string   `json:"sourceName"`
+	SourceHost     string   `json:"sourceHost"`
+	SourceCategory string   `json:"sourceCategory"`
+	Fields         fieldSet `json:"fields"`
+	Message        string   `json:"message"`
+}
+
+func (fields fieldSet) MarshalJSON() ([]byte, error) {
+	return json.Marshal(fields.attributes)
+}
+
+func transformLogsToIc(ld plog.Logs) ([]icRecord, error) {
+	icRecords := []icRecord{}
+	rls := ld.ResourceLogs()
+
+	for i := 0; i < rls.Len(); i++ {
+		rl := rls.At(i)
+		sourceCategoryVal := ""
+		sourceCategory, exists := rl.Resource().Attributes().Get(SourceCategoryKey)
+		if exists {
+			sourceCategoryVal = sourceCategory.AsString()
+		}
```

Comment:
Slightly more idiomatic way:

```suggestion
		if sourceCategory, exists := rl.Resource().Attributes().Get(SourceCategoryKey); exists {
			sourceCategoryVal = sourceCategory.AsString()
		}
```